### PR TITLE
language switcher refactor

### LINKED ISSUE #2238: login screen doesn't scroll
on small windows, buttons "below the fold" are inaccessible:

image

### PR DESCRIPTION
**Pull request by indirectlylit**
* Broke up language switcher into two components - modal and list. Pull common logic out into a mixin.
* Cleaned up language switcher modal, removing unnecessary local state and fixing A11Y issue
* Removed existing sign-up/sign-in language switcher footer, which had A11Y and responsiveness issues
* Factored out new language switcher list from setup wizard into a reusable component, and used that in sign-up and sign-in pages

fixes #2238

### CODE REVIEW

File: kolibri/core/assets/src/views/language-switcher/list.vue [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+    </select>
+  </div>
+
+</template>
+
+
+<script>
+
+  import { availableLanguages as allLanguages, currentLanguage } from 'kolibri.utils.i18n';
+  import { httpClient } from 'kolibri.client';
```
Comment (rtibbles) [inline kolibri/core/assets/src/views/language-switcher/list.vue]: Unused import.

File: kolibri/core/assets/src/views/language-switcher/mixin.js [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+export default {
+  methods: {
+    switchLanguage(code) {
+      const path = this.Kolibri.urls['kolibri:set_language']();
+      const entity = { language: code };
+      httpClient({
+        path,
+        entity,
+      }).then(() => {
+        global.location.reload(true);
```
Comment (rtibbles) [inline kolibri/core/assets/src/views/language-switcher/mixin.js:13]: Should we use the refresh browser global util here? Can't remember if it adds any value (I think the main reason it was added was for test isolation purposes, so can probably ignore).

File: kolibri/core/assets/src/views/language-switcher/modal.vue [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+  import kRadioButton from 'kolibri.coreVue.components.kRadioButton';
+  import languageSwitcherMixin from 'kolibri.coreVue.mixins.languageSwitcherMixin';
+  import { currentLanguage } from 'kolibri.utils.i18n';
+
+  export default {
+    name: 'languageSwitcherModal',
+    components: { coreModal, kButton, kRadioButton },
+    mixins: [languageSwitcherMixin],
+    $trs: {
+      changeLanguageModalHeader: 'Change language',
```
Comment (rtibbles) [inline kolibri/core/assets/src/views/language-switcher/modal.vue:40]: I assume the strings are unchanged? We will need to reupload strings and run TM after this, as the ids will have changed.

File: kolibri/plugins/setup_wizard/assets/src/views/onboarding-forms/default-language-form/index.vue [hunk 4]
```diff
@@ -85,55 +37,9 @@
...
-        httpClient({
-          path,
-          entity,
-        }).then(() => {
-          global.location.reload(true);
-        });
-      },
       setLanguage() {
-        this.submitDefaultLanguage(this.selectedLanguageId);
+        this.submitDefaultLanguage(this.currentLanguageId);
```
Comment (rtibbles) [inline kolibri/plugins/setup_wizard/assets/src/views/onboarding-forms/default-language-form/index.vue:42]: Where is currentLanguageId defined? I don't see it anywhere here, or any communication between the language list component and this component?

File: kolibri/plugins/user/assets/src/views/index.vue [hunk 5]
```diff
@@ -1,12 +1,16 @@
 <template>
 
   <div>
-    <core-base :navBarNeeded="navBarNeeded" :topLevelPageName="topLevelPageName" :appBarTitle="appBarTitle">
-      <component v-if="navBarNeeded" :is="currentPage"/>
+    <core-base
```
Comment (rtibbles) [inline kolibri/plugins/user/assets/src/views/index.vue]: Having `core-base` optionally rendered will break browser page title functionality.

Need to either reinstate it, or extract the page title functionality from core-base to a mixin and apply here in a special case (while also continuing to include it in core-base).

File: kolibri/plugins/setup_wizard/assets/src/views/onboarding-forms/default-language-form/index.vue (hunk 4 above)
Comment (indirectlylit) [reply kolibri/plugins/setup_wizard/assets/src/views/onboarding-forms/default-language-form/index.vue:42]: it's defined below as a vuex getter: https://github.com/learningequality/kolibri/pull/2259/files/20b9bdadc517763cb036fadca183c7989e5a7bf3#diff-4ff2174816eb01fe1a5db7055c66e187R51

(was hidden in github's default view of the diff)

File: kolibri/core/assets/src/views/language-switcher/mixin.js (hunk 2 above)
Comment (indirectlylit) [reply kolibri/core/assets/src/views/language-switcher/mixin.js:13]: >  the refresh browser global util 

I'm not sure what you're referring to

**Comment by indirectlylit** [comment]
@rtibbles I believe this should be good to go

File: kolibri/core/assets/src/views/language-switcher/mixin.js (hunk 2 above)
Comment (rtibbles) [reply kolibri/core/assets/src/views/language-switcher/mixin.js:13]: Oh, I guess it is a redirect util, not a refresh: https://github.com/learningequality/kolibri/blob/develop/kolibri/core/assets/src/utils/browser.js#L1

**Comment by rtibbles** [comment]
Codewise looks good to my eye, testing now.

**Pull request merged**